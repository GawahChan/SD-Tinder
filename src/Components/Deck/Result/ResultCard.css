.result-card {
    border-radius: 10px;
    box-shadow: 1px 1px 8px #333;
    margin: 10px;

    background-size: cover;
    background-position: top;
    overflow: hidden;

    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    
    min-width: 250px;
    min-height: 250px;
}

.result-card-title {
    color: white;
    background: linear-gradient(rgba(0,0,0,0), rgba(0,0,0,1));
    padding: 10px;
}


.result-card-title h2 {
    margin: 0;
    font-weight: 400;
}