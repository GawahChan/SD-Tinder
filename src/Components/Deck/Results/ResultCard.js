import React from 'react';
import {ResultProfile, ResultTitle} from './style';

function ResultCard(props) {
    return (
        <ResultProfile img={props.img}>
            <ResultTitle>{props.name}</ResultTitle>
        </ResultProfile>
    );
}

export default ResultCard;