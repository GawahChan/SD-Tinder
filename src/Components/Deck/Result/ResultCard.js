import React from 'react';
import './ResultCard.css';

function ResultCard(props) {
    return (
        <div className='result-card' style={{ 'backgroundImage': `url(${require(`../../../Common/Profile/${props.img}.jpg`)})` }}>
            <div className='result-card-title'>
                <h2>{props.name}</h2>
            </div>
        </div>
    );
}

export default ResultCard;