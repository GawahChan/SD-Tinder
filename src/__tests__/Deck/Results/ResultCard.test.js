import React from 'react';
import {shallow} from 'enzyme';
import ResultCard from '../../../Components/Deck/Results/ResultCard';

describe('ResultCard component', () => {
    const props = { img: 'Rosie-Mac', name: 'Rosie Mac'}
    const resultCard = shallow(<ResultCard {...props}/>)

    it('renders correctly', () => {
        expect(resultCard).toMatchSnapshot();
    });

})