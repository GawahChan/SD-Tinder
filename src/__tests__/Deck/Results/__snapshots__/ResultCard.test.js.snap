// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard component renders correctly 1`] = `
<styled.div
  img="Rosie-Mac"
>
  <Styled(styled.h1)>
    Rosie Mac
  </Styled(styled.h1)>
</styled.div>
`;
